lib/html-encoding-sniffer.js

```diff
@@ -37,11 +37,10 @@ function prescanMetaCharset(buffer) {
         i += 4;
         for (; i < l; i++) {
```

Comment:
Could we replace this whole thing with an indexOf('-->', start)?

Reply:
Oh, buffer is like a byte buffer?

Reply:
I doubt the perf will be that different but I like the idea; it's more elegant. I'll force-push a revised version.

Reply:
Still has an indexOf, so would work with Buffer. Do we support TypedArrays as well or something?

Reply:
Ah, it doesn't work because we're operating on a Buffer, not a string, so indexOf doesn't exist :(.

Reply:
Fascinating, I didn't realize Buffer had such niceties.

One day I would like to move all of jsdom away from Buffer and to pure Uint8Arrays, but this package definitely works on Buffers, so we can do that instead.

Reply:
Hmm it doesn't seem that much better. This is what I came up with:

```js
        i = buffer.indexOf("-->", i + 1);
        if (i === -1) {
          i = l;
          break;
        }
        i += 3;
```

Reply:
Could just break/return on the `=== -1`? But doesn't really change much.

I'm generally in favor of using the higher-level function if possible, even if they don't directly translate to a (substantical) savings in LOC, just to express intent better (and prevent bugs like this!).

Reply:
I'll keep it as-is; I think it's probably better to fit with the surrounding code style, which is rather bit-twiddly.